Dev

HaseProgram opened the pull request:
Ready for RK2

Commit:
Initial commit

init

HustonMmmavr committed:
Added Interface for views and fixed controllers. Still need understand connection to postgres!

HustonMmmavr committed:
Added Interface for views and fixed controllers. Still need understand connection to postgres!

HustonMmmavr committed:
db added

HustonMmmavr committed:
FIxed errors. Need to make CI.

HustonMmmavr committed:
Added inspection. TODO: cut project name; pom.xml fix (add project name, fix dependencies, park_checks run); CI; ReadMe; man; deploy.

Commit:
Added missing inspections

HustonMmmavr committed:
Added test fo signin

HustonMmmavr committed:
Revert "Added testing"

This reverts commit dc0669f15012c2b6c5dddf8c7d8da79c2f20334a.

HustonMmmavr committed:
Revert "Added testing"

This reverts commit dc0669f15012c2b6c5dddf8c7d8da79c2f20334a.

HustonMmmavr committed:
Added testing

HustonMmmavr committed:
Stable verson

HustonMmmavr committed:
deleted readme

HustonMmmavr committed:
f

HustonMmmavr committed:
d

HustonMmmavr committed:
trying deploy

HustonMmmavr committed:
trying to fix:

HustonMmmavr committed:
Last stable version

HustonMmmavr committed:
Stable version

HustonMmmavr committed:
fixed conflicts

HustonMmmavr committed:
trying to resolve

HustonMmmavr committed:
trying to resolve problem

added checkstyle

HustonMmmavr committed:
trying to delete old files

HustonMmmavr committed:
added new files

HustonMmmavr committed:
added crosses

HustonMmmavr committed:
fixed error with duplicated email and added tests

HustonMmmavr committed:
Added travis, fixed tests.

HustonMmmavr committed:
fixed properties

HustonMmmavr committed:
fixed travis

HustonMmmavr committed:
added service postgres

HustonMmmavr committed:
fixed error

HustonMmmavr committed:
fixed error in travis

HustonMmmavr committed:
Fixed incorrect code in controller, fixed tests, start write docu,entation for api.

HustonMmmavr committed:
writing docmentation

HustonMmmavr committed:
Fixed uncionality, added readme, documentation created.

HustonMmmavr committed:
Cross modified

HustonMmmavr committed:
Cleared pom, fixed swagger. Fixed bug in signup.

HustonMmmavr committed:
Fixed checks and added annotations.

HustonMmmavr committed:
fixed controller

signup

HaseProgram committed:
Merge pull request #1 from LastUnion/master

Working server

isopov commented:
РК2 так РК2 - я только примененного PR к РК1 не вижу ;-)

isopov commented:
1. Почистите лишние зависимости - spring-boot-configuration-processor ? spring-security-web ?
2. Несколько `@JsonCreator` на одном классе работает по счастливой случайности, а вообще не должно (ResponseCode)
3. Создание схемы сделайте через flyway
4. В базу из приложения не ходите из-под postgres
5. Чтобы не приходилось определять локально переменную окружения или править application.properties, можно определить урл к базе как-то так - `spring.datasource.url=${JDBC_DATABASE_URL:jdbc:postgresql://localhost/lastunion}`
6. Зачем junit-категории?
7. Константы лучше определять как константы - то есть public|private static final String
8. В тестах не бойтесь `throws Exception` - там хоть все методы такими могут быть, оборачивать в RuntimeException - только лишние строчки кода
9. А вообще `@SuppressWarnings("ThrowInsideCatchBlockWhichIgnoresCaughtException")` - зря игнорируете - это очень правильное замечание Идеи к вашему коду
10. lastunion.application.dao.UserDAO.modifyUser(UserModel, UserModel) - перепишите на параметризованный запрос - собирать самому из строчек - нельзя. (saveUser & deleteUserByName тоже)
11. Что вы думаете о таких методах? зачем они нужны?
```
    private void executeQuery(String query) {
        jdbcTemplate.update(query);
    }

    private String makePasswordHash(@NotNull final String password) {
        return passwordEncoder().encode(password);
    }

    private boolean checkPassword(@NotNull final String password, @NotNull final String passwordHash) {
        return passwordEncoder().matches(password, passwordHash);
    }

```
12. `@CrossOrigin` - скопирован между контроллерами, вынесите хоть в константу аргумент этих аннотаций
13. `messageSource.getMessage("msgs.forbidden", null, Locale.ENGLISH)` - в чем смысл такой локализации? может удалим совсем? или подсказать, как довести до рабочего состояния? вы уверены, что фронтенд к этому готов?
14. `@Bean` аннотация может быть на методе в классе - спринговой конфигурации, а не в другом бине (то есть не в Service, а например в Application), а в сервисе уже браться из спрингового контекста (аргументом конструктора)
15. UserDAO тоже сделайте спринговым бином (например, аннотацией `@Repository`)
16. Переделайте работу с ошибками - `return false` - это плохая обработка любых ошибок с базой. Логирование где есть, без стектрейса - это тоже не пойдёт.
17. Комментарии в коде зачем? У вас часто кода после комментариев меньше, чем самих комментариев - лучше переименуйте методы, чтобы они были понятнее (может они и так понятны?), а комментарии удалите такие
18. Тесты сделайте транзакционными, чтобы они не влияли никак друг на друга, и оставляли после себя чистую базу (только может с созданной через flyway схемой)

isopov reviewed (changes requested):
Замечания в комментарии (иях)

HustonMmmavr committed:
refactoring1

HustonMmmavr committed:
Tests refactored

HustonMmmavr committed:
refactored fsdf sdf

HustonMmmavr committed:
fixed error cs

Commit:
добавлено автоматическое переключения языков

isopov commented:
ping

HaseProgram committed:
Merge pull request #3 from LastUnion/master

Fixed errors and refactored.

HaseProgram commented:
У нас человек ошибся и создал отдельный пулреквест, вместо внесения сюда новых правок

HaseProgram committed:
Revert "Fixed errors and refactored. "

HaseProgram committed:
Merge pull request #4 from java-park-mail-ru/revert-3-master

Revert "Fixed errors and refactored. " Wrong Branch

Commit:
merged

Commit:
travis

Commit:
travis.yml added

isopov commented:
Ну а ответы какие-то я получу на вопросы? Например про такую организацию тестов - в других пакетах, чем основные исходники с категориями, интеграциионными тестами через failsafe-плагин - зачем это всё?

KSolovyev commented:
отдельный пул реквест не надо. Коммиты автоматически попадают в пул реквест, когда вы коммитите их в ветку

isopov commented:
Кажется, опять не в ту ветку были коммиты - вы мне лично говорили, что удалили нестандартные конфиги для тество, а они всё еще тут.

HustonMmmavr committed:
Deleted unused plugin from pom; deleted categories; added tests; added constant for crossorigin.

isopov commented:
Нифига себе - `@CrossOrigin(origins = "${frontend_url}")` - такое работает?

В общем круто - много вы тестов наколбасили. 

Про set_score - я из консоли разработчика в браузере  или курлом не смогу себе начитить счёт?

HustonMmmavr commented:
Да, с таким заданием константы адреса фронта в application,properties работает (нашел такое решение на stackoverflow).
Насчет set_score решение было принято в ночь перед предзащитой, пока единственное к чему мы пришли, это перенести всю логику игры на сервер.
